ads/openadstream.js

```diff
@@ -26,7 +26,7 @@ export function openadstream(global, data) {
   checkData(data, ['adhost', 'sitepage', 'pos', 'query']);
   //required fields
   validateDataExists(data, ['adhost', 'sitepage', 'pos']);
-  let url = 'https://' + encodeURIComponent(data.adhost) + '/3/' + encodeURIComponent(data.sitepage) + '/1' + new String(Math.random()).substring(2, 11) + '@' + data.pos;
+  let url = 'https://' + data.adhost + '/3/' + data.sitepage + '/1' + new String(Math.random()).substring(2, 11) + '@' + data.pos;
```

Comment:
if you remove the new String, you can't make the substring op anymore on the Math.random result. So it's either that or transforming through some other way to string.

The encodeURIComponent is not needed as well for the host, as it can causes same issue than the one for the data.sitepage.